pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17180786?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17180786?buildTab=artifacts#/) on master @ [a80bdf29d8f49844ed057fd47e243d1b396b1742](https://github.com/cockroachdb/cockroach/commits/a80bdf29d8f49844ed057fd47e243d1b396b1742):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/b8530b12366582f814734775ba1b4927/logTestTenantLogic_synthetic_privileges3249197906
    test_log_scope.go:76: use -show-logs to present logs inline
[23:30:21] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15939/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 60 statements
[23:30:27] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15939/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 68 statements
    logic.go:3034: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15939/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:305: SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=609b7e70 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/tables" iso=Serializable pri=0.00521964 epo=0 ts=1728084621.702265729,0 min=1728084621.702265729,0 seq=4} lock=true stat=PENDING rts=1728084621.702265729,0 wto=false gul=1728084622.202265729,0
[23:30:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15939/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4220: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15939/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:318: error while processing
    logic.go:4220: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15939/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:318: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b8530b12366582f814734775ba1b4927/logTestTenantLogic_synthetic_privileges3249197906
--- FAIL: TestTenantLogic_synthetic_privileges (10.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42781

--- a/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
+++ b/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
@@ -289,10 +289,10 @@ statement ok
 CREATE USER testuser4
 
 statement ok
-REVOKE SELECT ON crdb_internal.tables FROM public
+REVOKE SELECT ON crdb_internal.feature_usage FROM public
 
 query B retry
-SELECT has_table_privilege('testuser4', 'crdb_internal.tables', 'SELECT')
+SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
 ----
 false
 
@@ -300,24 +300,24 @@ statement ok
 BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
 
 statement ok
-GRANT SELECT ON crdb_internal.tables TO testuser4
+GRANT SELECT ON crdb_internal.feature_usage TO testuser4
 
 query TTTT
 SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
 ----
-public     /vtable/crdb_internal/tables  {}                                         {}
-root       /global/                      {MODIFYCLUSTERSETTING}                     {}
-testuser   /externalconn/foo             {USAGE}                                    {}
-testuser   /global/                      {MODIFYCLUSTERSETTING}                     {}
-testuser   /vtable/crdb_internal/tables  {SELECT}                                   {}
-testuser2  /externalconn/foo             {USAGE}                                    {}
-testuser2  /global/                      {MODIFYCLUSTERSETTING}                     {}
-testuser3  /global/                      {EXTERNALCONNECTION,MODIFYCLUSTERSETTING}  {}
-testuser4  /vtable/crdb_internal/tables  {SELECT}                                   {}
+public     /vtable/crdb_internal/feature_usage  {}                                         {}
+root       /global/                             {MODIFYCLUSTERSETTING}                     {}
+testuser   /externalconn/foo                    {USAGE}                                    {}
+testuser   /global/                             {MODIFYCLUSTERSETTING}                     {}
+testuser   /vtable/crdb_internal/tables         {SELECT}                                   {}
+testuser2  /externalconn/foo                    {USAGE}                                    {}
+testuser2  /global/                             {MODIFYCLUSTERSETTING}                     {}
+testuser3  /global/                             {EXTERNALCONNECTION,MODIFYCLUSTERSETTING}  {}
+testuser4  /vtable/crdb_internal/feature_usage  {SELECT}                                   {}
 
 # This should not cache the uncommitted privilege.
 query B
-SELECT has_table_privilege('testuser4', 'crdb_internal.tables', 'SELECT')
+SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
 ----
 true
 
@@ -327,17 +327,17 @@ ROLLBACK
 query TTTT
 SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
 ----
-public     /vtable/crdb_internal/tables  {}                                         {}
-root       /global/                      {MODIFYCLUSTERSETTING}                     {}
-testuser   /externalconn/foo             {USAGE}                                    {}
-testuser   /global/                      {MODIFYCLUSTERSETTING}                     {}
-testuser   /vtable/crdb_internal/tables  {SELECT}                                   {}
-testuser2  /externalconn/foo             {USAGE}                                    {}
-testuser2  /global/                      {MODIFYCLUSTERSETTING}                     {}
-testuser3  /global/                      {EXTERNALCONNECTION,MODIFYCLUSTERSETTING}  {}
+public     /vtable/crdb_internal/feature_usage  {}                                         {}
+root       /global/                             {MODIFYCLUSTERSETTING}                     {}
+testuser   /externalconn/foo                    {USAGE}                                    {}
+testuser   /global/                             {MODIFYCLUSTERSETTING}                     {}
+testuser   /vtable/crdb_internal/tables         {SELECT}                                   {}
+testuser2  /externalconn/foo                    {USAGE}                                    {}
+testuser2  /global/                             {MODIFYCLUSTERSETTING}                     {}
+testuser3  /global/                             {EXTERNALCONNECTION,MODIFYCLUSTERSETTING}  {}
 
 query B
-SELECT has_table_privilege('testuser4', 'crdb_internal.tables', 'SELECT')
+SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
 ----
 false
 

